project-koku/koku
# [COST-3929] replace OCP infra-type with correct infra-provider-type

## Jira Ticket

[COST-3929](https://issues.redhat.com/browse/COST-3929)

## Description

* during `get_ocp_infrastructure_map_trino`, JOIN on the reporting_tenant_api_provider table to get the correct infra type
* delete old entries in the ProviderInfrastructureMap where the infra type is OCP (this is not a valid infra type) 

## Testing

1. on main, ingest aws source followed by ocp-on-aws source
2. see in the db:
an incorrect infra-type:
```
postgres=# select * from api_providerinfrastructuremap;
 id | infrastructure_type |      infrastructure_provider_id      
----+---------------------+--------------------------------------
 17 | OCP                 | 2080ccb5-949d-4e44-962c-91d22f251d6b
```
and missing ocp-on-cloud costs:
```
postgres=# select count(*), sum(infrastructure_raw_cost), cluster_id from org1234567.reporting_ocpusagelineitem_daily_summary group by cluster_id;
 count | sum |    cluster_id    
-------+-----+------------------
   855 |     | my-ocp-cluster-1
```
4. checkout branch
5. create new data for the sources and download the data
6. check the db again:
see correct infra-type:
```postgres=# select * from api_providerinfrastructuremap;
 id | infrastructure_type |      infrastructure_provider_id      
----+---------------------+--------------------------------------
 20 | AWS-local           | 8b5a3eb0-8152-4635-bca3-651cbec080ac
```
and ocp-on-cloud costs:
```
postgres=# select count(*), sum(infrastructure_raw_cost), cluster_id from org1234567.reporting_ocpusagelineitem_daily_summary group by cluster_id;
 count |         sum         |    cluster_id    
-------+---------------------+------------------
  1941 | 307.318110800000006 | my-ocp-cluster-1
```
